tests/e2e/unified_test.go

```diff
@@ -443,6 +443,39 @@ func runScenario(ctx context.Context, t *testing.T, testPause bool, fixture reso
 					}
 				}
 
+				if ShouldTestMetadataLabels(t, testName, primaryResource) {
```

Comment:
I think I prefer doing this as a separate step to automagically changing the "main" reconciliation, but I also think we can just update create.yaml and update.yaml to include labels (at least one per kind)